webapi/src/main/scala/org/knora/webapi/messages/v1/responder/permissionmessages/PermissionMessagesV1.scala

```diff
@@ -0,0 +1,801 @@
+                    defaultObjectAccessPermissionsPerProject = Map.empty[IRI, Set[PermissionV1]] // remove default object access permission information
+                )
+            }
+            case PermissionDataType.FULL => {
+
+                PermissionDataV1(
+                    //projectInfos = projectInfos,
+                    groupsPerProject = groupsPerProject,
+                    administrativePermissionsPerProject = administrativePermissionsPerProject,
+                    defaultObjectAccessPermissionsPerProject = defaultObjectAccessPermissionsPerProject
+                )
+            }
+            case _ => throw BadRequestException(s"The requested userProfileType: $permissionProfileType is invalid.")
+        }
+    }
+
+    /* Is the user a member of the SystemAdmin group */
+    def isSystemAdmin: Boolean = {
+        groupsPerProject.contains(OntologyConstants.KnoraBase.SystemProject)
+    }
+
+    /* Does the user have the 'ProjectAllAdmin' permission for the project */
+    def hasProjectAdminAllPermissionFor(projectIri: IRI): Boolean = {
+        administrativePermissionsPerProject.get(projectIri) match {
+            case Some(permissions) => {
+                permissions(PermissionV1.ProjectAdminAllPermission)
+            }
+            case None => false
+        }
+    }
+
+
+    /*  */
+    /**
+      * Given an operation, checks if the user is allowed to perform it.
+      * @param operation the name of the operation.
+      * @param insideProject the IRI of the project inside which the operation will be performed.
+      * @param explain an optional parameter which defaults to false, when set to true, this method prints out an explanation for the result.
+      * @return a boolean value.
+      */
+    def hasPermissionFor(operation: OperationV1, insideProject: IRI, objectAccessPermissions: Option[Set[PermissionV1]], explain: Boolean = false): Boolean = {
+
+        //println(s"hasPermissionFor - administrativePermissionsPerProject: ${administrativePermissionsPerProject}, operation: $operation, insideProject: $insideProject")
+
+        if (this.isSystemAdmin) {
+            /* A member of the SystemAdmin group is allowed to perform any operation */
+            if (explain) println("TRUE: A member of the SystemAdmin group is allowed to perform any operation")
+            true
+        } else {
+            operation match {
+                case ResourceCreateOperation(resourceClassIri) => {
+                    this.administrativePermissionsPerProject.get(insideProject) match {
+                        case Some(set) => {
+                            set(PermissionV1.ProjectResourceCreateAllPermission)
+                        }
+                        case None => {
+                            if (explain) println("FALSE: No administrative permissions defined for this project.")
```

Comment:
I removed ``explain``, since it didn't do everything I wanted it to do anyway. Maybe I will add this at a later time.